packages/midi/src/MidiSetting.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react'
+import { HedronEngine, Input, NodeTypes, ParamWithInfo, UseEngineStore } from '@hedron/engine'
+import { Midi } from '.'
+
+interface IProps {
+  param: ParamWithInfo
+  engine: HedronEngine
+  useEngineStore: UseEngineStore
+  midi: Midi
+}
+
+/**
+ * Get the MIDI setting jsx element
+ * @param param The parameter to display the midi setting for
+ * @param engine hedron engine
+ * @param useEngineStore useEngineStore hook
+ * @param midi midi plugin
+ * @returns A JSX Elelemnt that displays connected midi inputs, and allows for midi learning
+ */
+export function getMidiSetting(
+  param: ParamWithInfo,
+  engine: HedronEngine,
+  useEngineStore: UseEngineStore,
+  midi: Midi,
+): JSX.Element {
+  return <MidiSetting param={param} engine={engine} useEngineStore={useEngineStore} midi={midi} />
+}
+
+/**
+ * Get the inputs that are connected to the node
+ * @param nodeId The node id to get the inputs for
+ * @param useEngineStore useEngineStore hook
+ * @returns The inputs that are connected to the node
+ */
+const useInputsWithNode = (nodeId: string, useEngineStore: UseEngineStore) => {
+  const inputs = useEngineStore((state) => state.inputs)
+  return Object.values(inputs).filter((input) => input.targetNodeIds.includes(nodeId))
```

Comment:
This is grabbing all inputs, might want a better way to register inputs to the plugin, so it can just get it's own inputs

Reply:
Hmm we have a duplicate hook inside `/hooks`. We're definitely going to want a separate package at some point soon for useful things like this! See my above comment about how to optimise this hook.